### PR TITLE
Round 1 of XML commenting

### PR DESCRIPTION
**Pull request by s3364859**
@6nop @s3364942 @s3494110 @s3482247 

In this PR:
- XML commenting for CPT331.Core, CPT331.Data, CPT331.Data.Parsers
- Some tidying where applicable (OCD)

boxing

### CODE REVIEW

File: CPT331.Core/ApplicationConfiguration.cs [hunk 1]
```diff
@@ -27,7 +33,10 @@ public static ApplicationConfiguration Default
 			}
 		}
 
-		[ApplicationScopedSetting()]
+		/// <summary>
+		/// Gets the connection string to the database.
+		/// </summary>
+		[ApplicationScopedSetting]
```
Comment (s3364859) [inline CPT331.Core/ApplicationConfiguration.cs]: OCD (I should really get that looked at).

File: CPT331.Core/Extensions/StringExtensions.cs [hunk 2]
```diff
@@ -8,24 +8,39 @@
 
 namespace CPT331.Core.Extensions
 {
+	/// <summary>
+	/// Provides System.String type extensions.
+	/// </summary>
 	public static class StringExtensions
 	{
-		private static SHA1CryptoServiceProvider _sha1CryptoServiceProvider;
```
Comment (s3364859) [inline CPT331.Core/Extensions/StringExtensions.cs]: Decided that this didn't need to live the potentially the life of the application.

File: CPT331.Core/ObjectModel/Coordinate.cs [hunk 3]
```diff
@@ -98,12 +119,6 @@ public static bool IsLongitudeValid(double longitude)
 			);
 		}
 
-        [Obsolete("Use IsValid(double, double) instead.")]
```
Comment (s3364859) [inline CPT331.Core/ObjectModel/Coordinate.cs]: Thanks for marking this - removed now.

File: CPT331.Core/ObjectModel/Coordinate.cs [hunk 4]
```diff
@@ -153,14 +168,12 @@ public static Coordinate FromValues(double latitude, double longitude)
...
-        /// </summary>
-        /// <returns>The string representation of the value of this instance.</returns>
+		/// <summary>
+		/// Converts the coordinate values to an equivalent string representation.
+		/// </summary>
+		/// <returns>The string representation of the value of this instance.</returns>
 		public override string ToString()
 		{
-            int x = 1;
-            x.ToString();
```
Comment (s3364859) [inline CPT331.Core/ObjectModel/Coordinate.cs]: Not sure what this bit was doing?

File: CPT331.Core/ObjectModel/EventFinderTransform.cs [hunk 5]
```diff
@@ -1,18 +0,0 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace CPT331.Core.ObjectModel
-{
-    public enum EventFinderImageFormat : int
```
Comment (s3364859) [inline CPT331.Core/ObjectModel/EventFinderTransform.cs]: Renamed above.

File: CPT331.Data.Migration/Migrations/M0002.sql [hunk 6]
```diff
@@ -727,7 +727,6 @@ INSERT INTO [Location].LocalGovernmentArea (StateID, Name) VALUES (@@StateID, 'P
 INSERT INTO [Location].LocalGovernmentArea (StateID, Name) VALUES (@@StateID, 'South East Metropolitan District')
 INSERT INTO [Location].LocalGovernmentArea (StateID, Name) VALUES (@@StateID, 'South Metropolitan District')
 INSERT INTO [Location].LocalGovernmentArea (StateID, Name) VALUES (@@StateID, 'South West District')
---	INSERT INTO [Location].LocalGovernmentArea (StateID, Name) VALUES (@@StateID, 'Western Australia')		--	I doubt this is actually required - there's no associated crime data
```
Comment (s3364859) [inline CPT331.Data.Migration/Migrations/M0002.sql]: Goodbye to you!

File: CPT331.Data.Parsers/WaKmlParser.cs [hunk 7]
```diff
@@ -45,12 +57,15 @@ protected override void OnParse(string fileName, List<Coordinate> coordinates)
 				}
 				
 				coordinates.Clear();
-				
-				string[] coordinateLines = coordinateValues.Split(" ".ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
+
+				string[] coordinateLines = coordinateValues
```
Comment (s3364859) [inline CPT331.Data.Parsers/WaKmlParser.cs]: Fiddle a bit with each line in order to get it to parse correctly.

File: CPT331.Data/CrimeRepository.cs [hunk 8]
```diff
@@ -14,61 +14,113 @@
...
 {
+	/// <summary>
+	/// Represents a CrimeRepository type, used to manipulate local government area data.
+	/// </summary>
 	public static class CrimeRepository
 	{
+		/// <summary>
+		/// The Crime.spAddCrime stored procedure name.
+		/// </summary>
+		public const string CrimeSpAddCrime = "Crime.spAddCrime";
```
Comment (s3364859) [inline CPT331.Data/CrimeRepository.cs]: Concreted up every sproc Schema.spName in this assembly.

File: CPT331.Data/EventRepository.cs [hunk 9]
```diff
@@ -1,32 +0,0 @@
...
-using System.Collections.Generic;
-
-using CPT331.Core.ObjectModel;
-using CPT331.WebAPI.Parsers;
-
-#endregion
-
-namespace CPT331.Data
-{
-	public static class EventRepository
```
Comment (s3364859) [inline CPT331.Data/EventRepository.cs]: Renamed above.

File: CPT331.Core/ObjectModel/Coordinate.cs (hunk 4 above)
Comment (6nop) [inline CPT331.Core/ObjectModel/Coordinate.cs]: Whoops.. I was trying to get to the built-in c# Meta Data the lazy way. I forgot to delete it afterwards.

/uppercut

**Comment by 6nop** [comment]
All good. Merging the PR.

**Pull request merged**